tests/benchmarks/power_consumption/temperature_sensor/src/main.c

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2024, Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+#include <zephyr/device.h>
+#include <zephyr/drivers/sensor.h>
+
+static const struct device *temp_dev = DEVICE_DT_GET(DT_NODELABEL(temp_sensor));
+
+void set_trigger(void);
+
+static void trigger_handler(const struct device *temp_dev, const struct sensor_trigger *trig)
+{
+	ARG_UNUSED(temp_dev);
+	ARG_UNUSED(trig);
+
+	k_busy_wait(1000000);
+	set_trigger();
+}
+
+void set_trigger(void)
+{
+	int rc;
+	struct sensor_value val = {0};
+	struct sensor_trigger trig = {.type = SENSOR_TRIG_THRESHOLD, .chan = SENSOR_CHAN_DIE_TEMP};
+
+	/* Set sampling frequency to 10 Hz, to expect a trigger after 100 ms. */
+	val.val1 = 1;
+	rc = sensor_attr_set(temp_dev, SENSOR_CHAN_DIE_TEMP, SENSOR_ATTR_SAMPLING_FREQUENCY, &val);
+	__ASSERT_NO_MSG(rc == 0);
+
+	rc = sensor_sample_fetch_chan(temp_dev, SENSOR_CHAN_DIE_TEMP);
+	__ASSERT_NO_MSG(rc == 0);
+
+	rc = sensor_channel_get(temp_dev, SENSOR_CHAN_DIE_TEMP, &val);
+	__ASSERT_NO_MSG(rc == 0);
+
+	printk("Temperature: %d.%d\n", val.val1, val.val2 / 10000);
```

Comment:
Please verify is temperature is in room range, like here: https://github.com/zephyrproject-rtos/zephyr/commit/fb86c453ccf0d65e808744719a1cd92eb78d1086